odxtools/exceptions.py

```diff
@@ -38,11 +32,13 @@ def odxraise(message: Optional[str] = None, error_type: Type[Exception] = OdxErr
 
     Also, convince type checkers that the exception is always raised.
     """
-    if TYPE_CHECKING or strict_mode:
-        if message is None:
-            raise error_type()
-        else:
-            raise error_type(message)
+    if TYPE_CHECKING:
+        raise error_type()
+
+    if message is None:
+        warnings.warn("ODX conformity error detected", error_type, stacklevel=2)
+    else:
+        warnings.warn(message, error_type, stacklevel=2)
```

Comment:
Wrappers of `odxraise`, such as `odxassert` and `odxrequire` need a different stacklevel=3